go/vt/sqlparser/parse_test.go

```diff
@@ -452,12 +452,48 @@ var (
 			input: "select /* keyword table alias */ 1 from t as `By`",
 		}, {
 			input: "select /* use */ 1 from t1 as of '2019-01-01' use index (a) where b = 1",
+		}, {
+			input: "select /* use */ 1 from t1 for system_time as of '2019-01-01' use index (`By`) where b = 1",
+			output: "select /* use */ 1 from t1 as of '2019-01-01' use index (`By`) where b = 1",
+		}, {
+			input: "select /* use */ 1 from t1 for system_time from '2019-01-01' to '2020-01-01' use index (`By`) where b = 1",
+			output: "select /* use */ 1 from t1 for system_time from '2019-01-01' to '2020-01-01' use index (`By`) where b = 1",
+		}, {
+			input: "select /* use */ 1 from t1 for system_time between '2019-01-01' and '2020-01-01' use index (`By`) where b = 1",
+			output: "select /* use */ 1 from t1 for system_time between '2019-01-01' and '2020-01-01' use index (`By`) where b = 1",
+		}, {
+			input: "select /* use */ 1 from t1 for system_time contained in ('2019-01-01', '2020-01-01') use index (`By`) where b = 1",
+			output: "select /* use */ 1 from t1 for system_time contained in ('2019-01-01', '2020-01-01') use index (`By`) where b = 1",
+		}, {
+			input: "select /* use */ 1 from t1 for system_time all use index (`By`) where b = 1",
+			output: "select /* use */ 1 from t1 for system_time all use index (`By`) where b = 1",
+		}, {
+			input: "select /* use */ 1 from t1 for version as of '2019-01-01' use index (`By`) where b = 1",
+			output: "select /* use */ 1 from t1 as of '2019-01-01' use index (`By`) where b = 1",
+		}, {
+			input: "select /* use */ 1 from t1 for version from '2019-01-01' to '2020-01-01' use index (`By`) where b = 1",
```

Comment:
(minor) I think `versions` reads a little better with the ranges, since you really are selecting multiple versions of the table. Plus, seems nice to use the standard terminology where we can.

Reply:
`for version` =~ `for system_time`, no?

Is `versions` a standard?

Reply:
Looks like SQL2011 defined "<table-name> VERSIONS BETWEEN ...". Seems fine to slightly alter the syntax here (i.e. "FOR VERSIONS BETWEEN"), like all vendors seem to do, just saying that "VERSIONS" seems more accurate if the query is using `BETWEEN` since it gives a better cue that multiple versions may be returned. I know you don't have to use `BETWEEN` in all queries though.